tests/misc-utils.c

```diff
@@ -0,0 +1,295 @@
+             GError **error)
+{
+  return g_file_replace_contents (path,
+                                  (bytes != NULL) ? g_bytes_get_data (bytes, NULL) : "",
+                                  (bytes != NULL) ? g_bytes_get_size (bytes) : 0,
+                                  NULL,
+                                  FALSE,
+                                  G_FILE_CREATE_NONE,
+                                  NULL,
+                                  NULL,
+                                  error);
+}
+
+gboolean
+create_directory (GFile *path,
+                  GError **error)
+{
+  g_autoptr(GError) local_error = NULL;
+
+  if (!g_file_make_directory_with_parents (path, NULL, &local_error))
+    if (!g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_EXISTS))
+      {
+        g_propagate_error (error, g_steal_pointer (&local_error));
+        return FALSE;
+      }
+
+  return TRUE;
+}
+
+gboolean
+create_symlink (const gchar *target,
+                GFile *link,
+                GError **error)
+{
+  g_autofree gchar *path = g_file_get_path (link);
+
+  if (symlink (target, path) < 0)
+    {
+      int saved_errno = errno;
+
+      g_set_error_literal (error, G_IO_ERROR, g_io_error_from_errno (saved_errno),
+                           g_strerror (saved_errno));
+
+      return FALSE;
+    }
+
+  return TRUE;
+}
+
+gboolean
+load_key_file (GFile *file,
+               GKeyFile **out_keyfile,
+               GError **error)
+{
+  g_autofree gchar *path = g_file_get_path (file);
+  g_autoptr(GKeyFile) keyfile = g_key_file_new ();
+
+  if (!g_key_file_load_from_file (keyfile,
+                                  path,
+                                  G_KEY_FILE_NONE,
+                                  error))
+    return FALSE;
+
+  *out_keyfile = g_steal_pointer (&keyfile);
+  return TRUE;
+}
+
+gboolean
+save_key_file (GFile *file,
+               GKeyFile *keyfile,
+               GError **error)
+{
+  g_autofree gchar *path = g_file_get_path (file);
+
+  return g_key_file_save_to_file (keyfile, path, error);
+}
+
+static GDateTime *
+get_now (void)
+{
+  static GDateTime *now = NULL;
+
+  if (now == NULL)
+    now = g_date_time_new_now_utc ();
+
+  return g_date_time_ref (now);
+}
+
+GDateTime *
+days_ago (gint days)
+{
+  g_autoptr(GDateTime) now = get_now ();
+  g_autoptr(GDateTime) now_fixed = g_date_time_new_utc (g_date_time_get_year (now),
+                                                        g_date_time_get_month (now),
+                                                        g_date_time_get_day_of_month (now),
+                                                        12,
+                                                        0,
+                                                        0);
+
+  return g_date_time_add_days (now_fixed, -days);
+}
+
+gboolean
+input_stream_to_string (GInputStream *stream,
+                        gchar **out_str,
+                        GError **error)
+{
+  gsize len = 2 * 1024 * 1024;
```

Comment:
Ah, just a length of a buffer which will contain the bytes read from the stream and put into the `GString`. I never know what is the golden number for this case.